### PR TITLE
Clarify Airflow variable usage on Astro

### PR DESCRIPTION
**Pull request by jwitz**
Resolves https://github.com/astronomer/docs/issues/1848 by consolidating our two "Use variables in Airflow" section into one. Also reframed the conversation as "Here's how you store Airflow variables on Astronomer" vs. "Here's how you use Astronomer environment variables in Airflow", because that seems like the framing most people will enter the doc with.

### CODE REVIEW

File: astro/environment-variables.md [hunk 1]
```diff
@@ -156,14 +147,18 @@ Airflow connections set with environment variables do not appear in the Airflow
...
+
+You can then use the following Python functions to fetch the variable value in the top level of your DAG code:
+
+- `Variable.get('<VAR_NAME>')`: This method is more secure for fetching secret values. However, this method can inhibit performance because it makes a request Airflow metadata database every time your DAGs are parsed, which can occur every 30 seconds. See [DAG writing best practices](https://docs.astronomer.io/learn/dag-best-practices#avoid-top-level-code-in-your-dag-file) for more information about avoiding repeated requests in top level code.
+- `os.getenv('AIRFLOW_VAR_<VAR_NAME>','<default-value>')`: This method is faster because it reduces the number of Airflow metadata database requests. However, it's less secure. Astronomer does not recommend using `os.getenv` with secret values because calling these values with the function can print them to your logs. 
+
+    Replace `default_value` a default value to use if Airflow can't find the environment variable. Typically, this is the value you defined for the environment variable in the Cloud UI.
```
Comment (jwitz) [inline astro/environment-variables.md:164]:
```suggestion
    Replace `<default_value>` with a default value to use if Airflow can't find the environment variable. Typically, this is the value you defined for the environment variable in the Cloud UI. 
```

File: astro/environment-variables.md [hunk 2]
```diff
@@ -156,14 +147,18 @@ Airflow connections set with environment variables do not appear in the Airflow
...
+Use Astro environment variables to store values you would normally store as [Airflow variables](https://airflow.apache.org/docs/apache-airflow/stable/howto/index.html) to have better control of custom values and configurations. To fetch a Deployment environment variable value from a DAG, you must format the variable key as `AIRFLOW_VAR_<VAR_NAME>`.
 
 For example, consider the following Airflow variable:
 
 - Variable name: `My_Var`
 - Value: `2`
 
 To store this Airflow variable as an environment variable, you create an environment variable with the key `AIRFLOW_VAR_MY_VAR` and the value `2`.
+
+You can then use the following Python functions to fetch the variable value in the top level of your DAG code:
```
Comment (kwcanuck) [inline astro/environment-variables.md:159]:
```suggestion
You can then use the following Python functions in the top level of your DAG code to fetch the variable value:
```

File: astro/environment-variables.md [hunk 3]
```diff
@@ -156,14 +147,18 @@ Airflow connections set with environment variables do not appear in the Airflow
...
+
+- `Variable.get('<VAR_NAME>')`: This method is more secure for fetching secret values. However, this method can inhibit performance because it makes a request Airflow metadata database every time your DAGs are parsed, which can occur every 30 seconds. See [DAG writing best practices](https://docs.astronomer.io/learn/dag-best-practices#avoid-top-level-code-in-your-dag-file) for more information about avoiding repeated requests in top level code.
```
Comment (kwcanuck) [inline astro/environment-variables.md:161]:
```suggestion
- `Variable.get('<VAR_NAME>')`: This method is more secure for fetching secret values. However, this method can affect performance because it makes a request to the Airflow metadata database every time your DAGs are parsed, which can occur every 30 seconds. See [DAG writing best practices](https://docs.astronomer.io/learn/dag-best-practices#avoid-top-level-code-in-your-dag-file) for more information about avoiding repeated requests in top level code.
```

**Review by kwcanuck — APPROVED** [review]

File: astro/environment-variables.md [hunk 4]
```diff
@@ -19,6 +19,8 @@ An environment variable on Astro is a key-value configuration that is applied to
 - Integrate with Datadog or other third-party tooling to [export Deployment metrics](deployment-metrics.md#export-airflow-metrics-to-datadog).
 - Set [Airflow configurations](https://airflow.apache.org/docs/apache-airflow/stable/configurations-ref.html?), such as default timezone and maximum active runs per DAG.
 
+You can use Astro environment variables to configure the same keys and values that you would configure as [Airflow variables](https://airflow.apache.org/docs/apache-airflow/stable/howto/index.html). However, Astro environment variables are stored and managed differently than Airflow variables. See [How environment variables are stored on Astro](#how-environment-variables-are-stored-on-astro).
```
Comment (BasPH) [inline astro/environment-variables.md:22]: IMO this isn't telling the reader much but "read this other doc to understand". I would rephrase to say they're stored as env vars and refer to the other doc for more info. For example:

```suggestion
You can use Astro variables to configure the same keys and values that you would configure as [Airflow variables](https://airflow.apache.org/docs/apache-airflow/stable/howto/index.html). However, Astro variables are exposed as environment variables and managed differently than Airflow variables. For more information, see [How environment variables are stored on Astro](#how-environment-variables-are-stored-on-astro).
```

**Review by BasPH — COMMENTED** [review]
There's a bit of an undesired mix of terms. The Astro UI doesn't call these "Environment variables", only "Variables":
image

I think it's worth adding some clarity e.g. "Astro variables are exported as environment variables. This enables you to set Airflow configuration (`AIRFLOW__[SECTION]__[KEY]`), but also set Airflow connections (`AIRFLOW_CONN_[CONNID]`) and variables (`AIRFLOW_VAR_[KEY]`).".

File: astro/environment-variables.md [hunk 5]
```diff
@@ -19,6 +19,8 @@ An environment variable on Astro is a key-value configuration that is applied to
 - Integrate with Datadog or other third-party tooling to [export Deployment metrics](deployment-metrics.md#export-airflow-metrics-to-datadog).
 - Set [Airflow configurations](https://airflow.apache.org/docs/apache-airflow/stable/configurations-ref.html?), such as default timezone and maximum active runs per DAG.
 
+You can use Astro variables to configure the same keys and values that you would configure as [Airflow variables](https://airflow.apache.org/docs/apache-airflow/stable/howto/index.html). However, Astro variables are exposed as cluster level environment variables and managed differently than Airflow variables. For more information, see [How environment variables are stored on Astro](#how-environment-variables-are-stored-on-astro).
```
Comment (jwitz) [inline astro/environment-variables.md:22]:
```suggestion
You can use Astro variables to configure the same keys and values that you would configure as [Airflow variables](https://airflow.apache.org/docs/apache-airflow/stable/howto/index.html). However, Astro variables are exposed as environment variables and managed differently than Airflow variables. For more information, see [How environment variables are stored on Astro](#how-environment-variables-are-stored-on-astro).
```

File: astro/environment-variables.md [hunk 6]
```diff
@@ -19,6 +19,8 @@ An environment variable on Astro is a key-value configuration that is applied to
 - Integrate with Datadog or other third-party tooling to [export Deployment metrics](deployment-metrics.md#export-airflow-metrics-to-datadog).
 - Set [Airflow configurations](https://airflow.apache.org/docs/apache-airflow/stable/configurations-ref.html?), such as default timezone and maximum active runs per DAG.
 
+You can use Astro variables to configure the same keys and values that you would configure as [Airflow variables](https://airflow.apache.org/docs/apache-airflow/stable/howto/index.html). However, Astro variables are exposed as environment variables and managed differently than Airflow variables. For more information, see [How environment variables are stored on Astro](#how-environment-variables-are-stored-on-astro).
```
Comment (jwitz) [inline astro/environment-variables.md:22]:
```suggestion
You can use environment variables to configure the same keys and values that you would configure as [Airflow variables](https://airflow.apache.org/docs/apache-airflow/stable/howto/index.html). However, Astro variables are exposed as environment variables and managed differently than Airflow variables. For more information, see [How environment variables are stored on Astro](#how-environment-variables-are-stored-on-astro).
```
Replace Astro with "environment"

File: astro/environment-variables.md [hunk 7]
```diff
@@ -156,14 +147,18 @@ Airflow connections set with environment variables do not appear in the Airflow
 
 ### Airflow variables
 
-The environment variable naming convention for Airflow variables is:
-
-- Key: `AIRFLOW_VAR_<VAR_NAME>` 
-- Value: `<value>`
+Use Astro environment variables to store values you would normally store as [Airflow variables](https://airflow.apache.org/docs/apache-airflow/stable/howto/index.html) to have better control of custom values and configurations. To fetch a Deployment environment variable value from a DAG, you must format the variable key as `AIRFLOW_VAR_<VAR_NAME>`.
```
Comment (jwitz) [inline astro/environment-variables.md:150]:
```suggestion
Use environment variables to store values you would normally store as [Airflow variables](https://airflow.apache.org/docs/apache-airflow/stable/howto/index.html) to have better control of custom values and configurations. To fetch a Deployment environment variable value from a DAG, you must format the variable key as `AIRFLOW_VAR_<VAR_NAME>`.
```

File: astro/environment-variables.md [hunk 8]
```diff
@@ -156,14 +147,18 @@ Airflow connections set with environment variables do not appear in the Airflow
 
 ### Airflow variables
 
-The environment variable naming convention for Airflow variables is:
-
-- Key: `AIRFLOW_VAR_<VAR_NAME>` 
-- Value: `<value>`
+Use environment variables to store values you would normally store as [Airflow variables](https://airflow.apache.org/docs/apache-airflow/stable/howto/index.html) to have better control of custom values and configurations. To fetch a Deployment environment variable value from a DAG, you must format the variable key as `AIRFLOW_VAR_<VAR_NAME>`.
```
Comment (jwitz) [inline astro/environment-variables.md:150]:
```suggestion
Use environment variables to store values you would normally store as [Airflow variables](https://airflow.apache.org/docs/apache-airflow/stable/howto/index.html). This gives you better control and security of custom values. To fetch a Deployment environment variable value from a DAG, you must format the variable key as `AIRFLOW_VAR_<VAR_NAME>`.
```

File: astro/environment-variables.md [hunk 9]
```diff
@@ -19,6 +19,8 @@ An environment variable on Astro is a key-value configuration that is applied to
 - Integrate with Datadog or other third-party tooling to [export Deployment metrics](deployment-metrics.md#export-airflow-metrics-to-datadog).
 - Set [Airflow configurations](https://airflow.apache.org/docs/apache-airflow/stable/configurations-ref.html?), such as default timezone and maximum active runs per DAG.
 
+You can use environment variables to configure the same keys and values that you would configure as [Airflow variables](https://airflow.apache.org/docs/apache-airflow/stable/howto/index.html). However, environment variables are exposed as environment variables and managed differently than Airflow variables. For more information, see [How environment variables are stored on Astro](#how-environment-variables-are-stored-on-astro).
```
Comment (jwitz) [inline astro/environment-variables.md:22]:
```suggestion
You can use environment variables to configure the same keys and values that you would configure as [Airflow variables](https://airflow.apache.org/docs/apache-airflow/stable/howto/index.html). However, the variables that you configure in Astro are exposed as environment variables and managed differently than Airflow variables. For more information, see [How environment variables are stored on Astro](#how-environment-variables-are-stored-on-astro).
```
